src/services/taskService.ts

```diff
@@ -0,0 +1,87 @@
+import { useState, useEffect, Dispatch, SetStateAction } from "react";
+
+export interface Task {
+  id: number;
+  content: string;
+  completed: boolean;
+}
+
+interface TaskServiceProps {
+  items: Task[];
+}
+
+interface TaskServiceReturnType extends TaskServiceProps {
+  selectedButton: string | null;
+  setSelectedButton: Dispatch<SetStateAction<string | null>>;
+  pendingItems: Task[];
+  totalTasks: number;
+  completedTasks: number;
+  addItem: (content: string) => void;
+  deleteItem: (id: number) => void;
+  checkItem: (id: number) => void;
+}
+
+export const useTaskService = ({
```

Comment:
There have been several improvements to this function.
Apart from the improvements already mentioned above, it would be great if we had a global state (React Context, Redux and others) to manage tasks, as this would avoid prop drilling in different parts of the code.